File: app/src/main/java/com/twismart/wallpapershd/ui/main/MainActivity.java
package com.twismart.wallpapershd.ui.main;

import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;

import com.twismart.wallpapershd.R;
import com.twismart.wallpapershd.ui.base.BaseActivity;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getActivityComponent().inject(this);
        setUp();
    }

    @Override
    protected void setUp() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        TabLayout tabLayout = (TabLayout) findViewById(R.id.tab_layout);
        tabLayout.setTabMode(TabLayout.MODE_FIXED);

        ViewPager viewPager = (ViewPager) findViewById(R.id.view_pager);
        viewPager.setAdapter(new MainPagerAdapter(getSupportFragmentManager(), MainActivity.this));
        tabLayout.setupWithViewPager(viewPager);
    }
}

File: app/src/main/kotlin/com/twismart/wallpapershd/ui/main/MainActivity.kt
/*
 * Copyright (C) 2017 Sneyder Angulo.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.twismart.wallpapershd.ui.main

import android.support.design.widget.TabLayout
import android.os.Bundle
import com.twismart.wallpapershd.R
import com.twismart.wallpapershd.ui.adapters.ZoomOutPageTransformer
import com.twismart.wallpapershd.ui.base.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        activityComponent.inject(this)
        setUp()
    }

    override fun setUp() {
        setSupportActionBar(toolbar)

        tabLayout.tabMode = TabLayout.MODE_FIXED

        mViewPager.adapter = MainPagerAdapter(fragmentManager, this@MainActivity)
        mViewPager.setPageTransformer(true, ZoomOutPageTransformer())
        tabLayout.setupWithViewPager(mViewPager)
    }
}